src/components/TopScrollingBar.tsx

```diff
@@ -0,0 +1,62 @@
+import React, { useState, useEffect, useRef, useCallback } from 'react';
+
+interface DoubleScrollbarProps {
+  children: React.ReactNode;
+}
+const TopScrollingBar: React.FC<DoubleScrollbarProps> = ({ children }) => {
+  const [width, setWidth] = useState<string>('auto');
+  const outerDivRef = useRef<HTMLDivElement | null>(null);
+  const childrenWrapperRef = useRef<HTMLDivElement | null>(null);
+
+  const getChildWrapperWidth = useCallback(() => {
+    const childrenWrapper = childrenWrapperRef.current;
+    const table = childrenWrapper?.getElementsByTagName('table')[0];
+    return table ? table.scrollWidth + 'px' : null;
+  }, []);
+
+  const setInnerDivWidth = useCallback(() => {
+    const newWidth = getChildWrapperWidth();
+    if (newWidth !== null && newWidth !== width) {
+      setWidth(newWidth);
+    }
+  }, [getChildWrapperWidth, width]);
+
+  useEffect(() => {
+    setInnerDivWidth();
+
+    const topScrollBarDiv = outerDivRef.current;
+    const tableWrapper: HTMLDivElement | null | undefined =
+      childrenWrapperRef.current?.querySelector('[datatype="table-wrapper"]');
+
+    if (!tableWrapper || !topScrollBarDiv) return;
+    const onTableWrapperScroll = () => {
+      tableWrapper.scrollLeft = topScrollBarDiv.scrollLeft;
+    };
+    topScrollBarDiv.addEventListener('scroll', onTableWrapperScroll);
+    const onTopBarScroll = () => {
+      topScrollBarDiv.scrollLeft = tableWrapper.scrollLeft;
+    };
+    tableWrapper.addEventListener('scroll', onTopBarScroll);
+
+    window.addEventListener('resize', setInnerDivWidth);
```

Comment:
just to remember for future work,  addEventListener is not recommended  to  use in React, but to use React's built-in event handling system, which works more effectively with React's virtual DOM and component lifecycle.
https://legacy.reactjs.org/docs/handling-events.html#:~:text=React%20events%20do%20not%20work,element%20after%20it%20is%20created.